daccs_client/client.py

```diff
@@ -31,94 +25,96 @@ def __init__(self, fallback: Optional[bool] = True) -> None:
             if the cloud registry cannot be accessed, defaults to True
         :type fallback: Optional[bool], optional
         :raises requests.exceptions.RequestException: Raised when there is an issue
-            conencting to the cloud registry and `fallback` is False
-        :raises UserWarning: Raised when there is an issue conencting to the cloud registry
+            connecting to the cloud registry and `fallback` is False
+        :raises UserWarning: Raised when there is an issue connecting to the cloud registry
             and `fallback` is True
         :raise RuntimeError: If cached registry needs to be read but there is no cache
         """
         self._fallback = fallback
-        self._using_fallback = False
         self._nodes: dict[str, DACCSNode] = {}
+        self._registry: dict = {}
 
         try:
-            self._registry = requests.get(REG_URL)
-            self._registry.raise_for_status()
-        except (requests.exceptions.RequestException, requests.exceptions.ConnectionError) as e:
-            if not self._fallback:
-                raise
-            else:
-                self._using_fallback = True
+            registry = requests.get(NODE_REGISTRY_URL)
+            registry.raise_for_status()
+        except (requests.exceptions.RequestException, requests.exceptions.ConnectionError):
+            if self._fallback:
                 warnings.warn("Cannot retrieve cloud registry. Falling back to cached version")
-
-        if not self._using_fallback:
-            self._load_registry_from_cloud()
+                self._load_registry_from_cache()
+            else:
+                raise
         else:
-            self._load_registry_from_cache()
+            self._load_registry_from_cloud(registry)
 
         for node, node_details in self._registry.items():
             self._nodes[node] = DACCSNode(node, node_details)
 
     @property
-    def nodes(self):
+    def nodes(self) -> dict[str, DACCSNode]:
         return self._nodes
 
-    def __getitem__(self, node):
+    def __getitem__(self, node: str) -> DACCSNode:
         try:
             return self.nodes[node]
         except KeyError:
             raise UnknownNodeError(f"No node named '{node}' in the DACCS network.") from None
 
-    def _load_registry_from_cloud(self):
+    def _load_registry_from_cloud(self, registry_response: requests.Response) -> None:
         try:
-            self._registry = self._registry.json()
-        except json.json.JSONDecodeError as e:
+            self._registry = registry_response.json()
+        except json.JSONDecodeError:
             raise RuntimeError(
                 "Could not parse JSON returned from the cloud registry. "
-                "Consider re-trying with 'fallback' set to True when instantiating the DACCS object."
+                f"Consider re-trying with 'fallback' set to True when instantiating the {self.__class__.__name__} "
```

Comment:
Not strictly necessary but it's nice so you have one fewer thing to change if the class name changes as well